src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp

```diff
@@ -0,0 +1,274 @@
+            intersecting_elements()
+        {
+            return std::move( intersecting_elements_ );
+        }
+
+    private:
+        bool triangle_edge_intersect(
+            geode::index_t triangle_id, geode::index_t edge_id ) const;
+
+    private:
+        const geode::TriangulatedSurface< dimension >& surface_;
+        const geode::EdgedCurve< dimension >& curve_;
+        std::vector< std::pair< geode::index_t, geode::index_t > >
+            intersecting_elements_;
+    };
+
+    template <>
+    bool TriangleEdgeIntersection< 2 >::triangle_edge_intersect(
+        geode::index_t triangle_id, geode::index_t edge_id ) const
+    {
+        const auto segment = curve_.segment( edge_id );
+        for( const auto e : geode::LRange{ 3 } )
+        {
+            const auto edge_vertices =
+                surface_.polygon_edge_vertices( { triangle_id, e } );
+            const geode::Segment2D edge{ surface_.point( edge_vertices[0] ),
+                surface_.point( edge_vertices[1] ) };
+            const auto result =
+                geode::segment_segment_intersection_detection( segment, edge );
+            if( result.first == geode::Position::inside )
+            {
+                return true;
+            }
+            if( result.second == geode::Position::inside )
+            {
+                return true;
+            }
+            if( result.first == geode::Position::parallel
+                && result.second == geode::Position::parallel )
+            {
+                const auto result_colinear =
+                    geode::colinear_segment_segment_intersection_detection(
+                        segment, edge );
+                if( result_colinear.first != geode::Position::outside
+                    || result_colinear.second != geode::Position::outside )
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    template <>
+    bool TriangleEdgeIntersection< 3 >::triangle_edge_intersect(
+        geode::index_t triangle_id, geode::index_t edge_id ) const
+    {
+        const auto triangle = surface_.triangle( triangle_id );
+        const auto segment = curve_.segment( edge_id );
+        const auto result =
+            geode::segment_triangle_intersection_detection( segment, triangle );
+        if( result.first == geode::Position::outside
+            || result.second == geode::Position::outside )
+        {
+            return false;
+        }
+
+        if( result.first == geode::Position::inside
+            || result.first == geode::Position::parallel )
+        {
+            return true;
+        }
+
+        if( result.second == geode::Position::inside
+            || result.second == geode::Position::edge0
+            || result.second == geode::Position::edge1
+            || result.second == geode::Position::edge2
+            || result.first == geode::Position::parallel )
+        {
+            return true;
+        }
```

Comment:
Are there cases where segment_triangle_inter return something else than outside and there are no intersections ?
If yes, leave as is, if not, leave only return true;